mm2src/mm2_bitcoin/script/src/builder.rs

```diff
@@ -39,12 +39,26 @@ impl Builder {
             .into_script()
     }
 
-    /// Builds p2wpkh or p2wsh script pubkey
-    pub fn build_p2witness(address: &AddressHashEnum) -> Script {
-        Builder::default()
-            .push_opcode(Opcode::OP_0)
-            .push_bytes(&address.to_vec())
-            .into_script()
+    /// Builds p2wpkh script pubkey
+    pub fn build_p2wpkh(address_hash: &AddressHashEnum) -> Result<Script, Error> {
+        match address_hash {
+            AddressHashEnum::AddressHash(wpkh_hash) => Ok(Builder::default()
+                .push_opcode(Opcode::OP_0)
+                .push_bytes(wpkh_hash.as_ref())
+                .into_script()),
+            _ => Err(Error::WitnessHashMismatched),
```

Comment:
Please match all enum variants to remember handling any added variants in the future.

Reply:
done